src/main/java/com/dataguise/saas/service/TaskBuilderService.java

```diff
@@ -0,0 +1,148 @@
+        } else if (taskType.contentEquals(PROTECTION_TYPE)) {
+            taskName.append("mt_" + getTimeStampForDgSecure());
+        }
+        return taskName.toString();
+    }
+
+    public String constructStructureName() {
+        StringBuffer structureName = new StringBuffer();
+        structureName.append("st_"+getTimeStampForDgSecure());
+        return structureName.toString();
+    }
+
+    public <T,T1> T1 addTaskBuilderPolicies(T taskBuilder) throws BadGatewayException, InternalServerException {
+        T1 builder = null;
+        SettingsDTO settingsDTO = settingsController.loadSettings();
+        List<String> policyList = settingsDTO.getPolicySelected();
+        if( taskBuilder instanceof DetectionTask.DetectionTaskBuilder) {
+            DetectionTask.DetectionTaskBuilder detectionTaskBuilder = ((DetectionTask.DetectionTaskBuilder) taskBuilder);
+            for(String policyType : policyList) {
+                if(policyType.equalsIgnoreCase(PolicyType.HIPAA.name())) {
+                    detectionTaskBuilder.addPolicyIds(1);
+                } else if (policyType.equalsIgnoreCase(PolicyType.PCI.name())) {
+                    detectionTaskBuilder.addPolicyIds(2);
+                } else if (policyType.equalsIgnoreCase(PolicyType.PII.name())) {
+                    detectionTaskBuilder.addPolicyIds(3);
+                } else if (policyType.equalsIgnoreCase(PolicyType.GDPR.name())) {
+                    detectionTaskBuilder.addPolicyIds(8);
+                } else {
+                    throw new InternalServerException(ErrorConstants.INVALID_POLICY_TYPE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
+            builder = (T1) detectionTaskBuilder;
+        } else if( taskBuilder instanceof MaskerTaskBuilder) {
+            MaskerTaskBuilder maskerTaskBuilder = ((MaskerTaskBuilder) taskBuilder);
+            for(String policyType : policyList) {
+                if(policyType.equalsIgnoreCase(PolicyType.HIPAA.name())) {
+                    maskerTaskBuilder.addPolicy(1);
+                } else if (policyType.equalsIgnoreCase(PolicyType.PCI.name())) {
+                    maskerTaskBuilder.addPolicy(2);
+                } else if (policyType.equalsIgnoreCase(PolicyType.PII.name())) {
+                    maskerTaskBuilder.addPolicy(3);
+                } else if (policyType.equalsIgnoreCase(PolicyType.GDPR.name())) {
+                    maskerTaskBuilder.addPolicy(8);
+                } else {
+                    throw new InternalServerException(ErrorConstants.INVALID_POLICY_TYPE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
+            builder = (T1) maskerTaskBuilder;
+        } else if( taskBuilder instanceof LFADetectionTask.LFADetectionTaskBuilder) {
+            LFADetectionTask.LFADetectionTaskBuilder LFADetectionTaskBuilder = ((LFADetectionTask.LFADetectionTaskBuilder) taskBuilder);
```

Comment:
**Reporter**: PMD    
    **Rule**: LocalVariableNamingConventions
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/TaskBuilderService.java L98
    
    The local variable name &#39;LFADetectionTaskBuilder&#39; doesn&#39;t match &#39;[a-z][a-zA-Z0-9]*&#39;&#10;&#10;&#10;Code Style https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#localvariablenamingconventions
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1204926880>*